src/app/components/secret-key/mnemonic-key/mnemonic-input-field.tsx

```diff
@@ -0,0 +1,85 @@
+import { useCallback, useState } from 'react';
+
+import { TextField } from '@radix-ui/themes';
+import { useField } from 'formik';
+import { css } from 'leather-styles/css';
+import { FlexProps, styled } from 'leather-styles/jsx';
+
+import { useIsFieldDirty } from '@app/common/form-utils';
+
+interface InputFieldProps extends FlexProps {
+  dataTestId?: string;
+  name: string;
+  value: string;
+  onPaste: (e: React.ClipboardEvent<HTMLInputElement>) => void;
+  onUpdateWord: (word: string) => void;
+  hasError?: boolean;
+  wordlist: string[];
+}
+export function InputField({ dataTestId, name, onPaste, onChange, value }: InputFieldProps) {
+  const [field, meta] = useField(name);
+  const [isFocused, setIsFocused] = useState(false);
```

Comment:
Checkout this hook which wraps focus so you don't need to handle state yourself https://github.com/leather-wallet/extension/blob/985e7dfe6594f1fe54ccd61b76d9f6606ca99092/src/app/components/item-hover.tsx#L47